### DISCUSSION TITLE
Does anyone have experience using MoltenVK with Docker Desktop?

### DISCUSSION THREAD

**Discussion by esba1ley** [KhronosGroup/MoltenVK#2127] [Q&A]
Hi -- Does anyone in this community have any experience accelerating 3D applications inside containers (e.g. Docker Desktop) using MoltenVK?  I assume the container needs Vulcan drivers, but how does the container connect to the host's hardware accelerated graphics?  the application I'm currently working with uses X11 (not Wayland) from inside a Docker container.

Any suggestions as to how to connect a 3D Graphics application built for Linux (x86) using the Vulcan drivers with mesa, connecting to MoltenVK would be very useful to my project.

**Comment by cdavis5e** [comment]
MoltenVK is for Apple Darwin (i.e. macOS, iOS, etc.), not Linux. Why, then, are you trying to use MoltenVK from a Linux container? Are you trying to spin up a Docker instance on macOS? (Is that even possible?)

**Reply by esba1ley** [reply]
Yes, I'm trying to run a Linux container that's hosting 3-D graphics through open GL or some other variant of software hosted on an M1 Apple machine. I'm trying to see if the metal environment can work as a backend for a Linux 3-D application that is hosted from docker desktop and yes, docker desktop works very well on macOS even running x86 builds emulated through Rosetta 2.
